web_external/js/models/SessionModel.js

```diff
@@ -42,99 +36,54 @@ minerva.models.SessionModel = girder.models.ItemModel.extend({
+     */
+    createSessionMetadata: function () {
+        var metadata = this.metadata() || {};
+        // TODO this is highly map centric, probably should be split out
+        var map = {};
+        map.basemap = 'osm';
+        map.basemap_args = {
             tileUrl: 'https://{s}.tiles.mapbox.com/v3/datamade.hn83a654/{z}/{x}/{y}.png',
             attribution: '<a href=http://www.mapbox.com/about/maps/ target=_blank>Terms &amp; Feedback</a>'
         };
-        sessionJsonContents.center = {x: -100, y: 36.5};
-        sessionJsonContents.zoom = 4;
-        sessionJsonContents.features = [];
-        // now save this as session.json as a file in the item
-        this.sessionJsonFile = new girder.models.FileModel();
-        this.sessionJsonFile.on('g:upload.complete', function () {
-            callback();
-        }, this).uploadToItem(this, JSON.stringify(sessionJsonContents), 'session.json', 'application/json');
-    },
-
-    fetch: function () {
-        this.on('g:fetched', function () {
-            // now get the session file
-            var sessionParams = {
-                path: 'minerva_session/' + this.get('_id') + '/session',
-                type: 'GET'
-            };
-            girder.restRequest(sessionParams).done(_.bind(function (file) {
-                this.sessionJsonFile = new girder.models.FileModel(file);
-                // TODO if there isn't one need to create it
-                // now we have the file, get the actual contents
-                girder.restRequest({
-                    path: 'file/' + file._id + '/download',
-                    type: 'GET'
-                }).done(_.bind(function (resp) {
-                    this.sessionJsonContents = resp;
-                    this.trigger('m:fetched');
-                }, this)).error(_.bind(function (err) {
-                    console.error(err);
-                    girder.events.trigger('g:alert', {
-                        icon: 'cancel',
-                        text: 'Could not download session json contents.',
-                        type: 'error',
-                        timeout: 4000
-                    });
-                }, this));
-            }, this)).error(_.bind(function (err) {
-                console.error(err);
-                girder.events.trigger('g:alert', {
-                    icon: 'cancel',
-                    text: 'Could not load session item.',
-                    type: 'error',
-                    timeout: 4000
-                });
-            }, this));
-        }, this).on('g:error', function () {
-            minerva.router.navigate('sessions', {trigger: true});
-        }, this);
-        girder.models.ItemModel.prototype.fetch.call(this);
+        map.center = {x: -100, y: 36.5};
+        map.zoom = 4;
+        map.features = [];
+        metadata.map = map;
+        this.on('minerva.metadata.saved', function () {
+            this.trigger('minerva.session.saved', this);
```

Comment:
>Is `minerva.` the new `m:`?

I was hoping for that, as once we start building many apps or many interacting repos it seemed like there was room for collision.  I'm fine with the `m:`.

Not long ago I had cataloged the Girder events and thought it was unclear from the name where many events were originating.  My overall goal was trying to provide more meaningful event names.

I wasn't aware of the jquery namespacing issues, thanks for explaining, though I don't think I understand yet.  I set up the following event handlers

    session.on('minerva.session.saved', function () { console.log('minerva.session.saved'); }, this);
    session.on('minerva.session', function () { console.log('minerva.session'); }, this);
    session.on('minerva.saved', function () { console.log('minerva.saved'); }, this);
    session.on('session.saved', function () { console.log('session.saved'); }, this);
    session.on('.session.saved', function () { console.log('.session.saved'); }, this);
    session.on('session', function () { console.log('session'); }, this);
    session.on('.session', function () { console.log('.session'); }, this);
    session.on('saved', function () { console.log('saved'); }, this);
    session.on('.saved', function () { console.log('.saved'); }, this);

and when I triggered `'minerva.session.saved'` only the `'minerva.session.saved'` handler reported back.  What am I missing here?